arch/arm/crc32_acle.c

```diff
@@ -5,17 +5,36 @@
  *
 */
 
-#ifdef __ARM_FEATURE_CRC32
-#include <arm_acle.h>
-#ifdef ZLIB_COMPAT
+#if __ARM_FEATURE_CRC32
```

Comment:
Was the change from ifdef to if intentional? Or was it supposed to be if defined()?

Reply:
This change was intentional to make it look as the other uses of that compiler defined macro.
It can be used both ways, either as `#if __ARM_FEATURE_CRC32` or `#if defined(__ARM_FEATURE_CRC32)` as both gcc and clang have it defined as `1` when compiling with `-march=armv8-a+crc`:
```
$ arm-linux-gnueabihf-gcc -march=armv8-a+crc -x c /dev/null -dM -E | grep __ARM_FEATURE_CRC32
#define __ARM_FEATURE_CRC32 1
```
and it is not defined when `-march` does not include crc:
```
$ arm-linux-gnueabihf-gcc -march=armv8-a -x c /dev/null -dM -E | grep __ARM_FEATURE_CRC32 | wc -l
0
```

Reply:
I guess that is fine then. My only worry is that it implies that we care about the value and not just whether it has been defined. For readability, I think just checking whether it has been defined would be more readable and easier to understand, but if you think this is better, I can be swayed.